docs/docs/icechunk-python/performance.md

```diff
@@ -318,3 +314,85 @@ Notice that the persisted config is restored when opening a Repository
 ```python exec="on" session="perf" source="material-block"
 print(ic.Repository.open(storage).config.manifest)
 ```
+
+## Preloading manifests
+
+While [manifest splitting](./performance.md#splitting-manifests) is a great way to control the size of manifests, it can be useful to configure the manner in which manifests are loaded. In Icechunk manifests are loaded lazily by default, meaning that when you read a chunk, Icechunk will only load the manifest for that chunk when it is needed to fetch the chunk data. While this is good for memory performance, it can be a bottleneck for read performance when you are reading a lot of chunks.
+
+To address this, Icechunk provides a way to preload manifests at the time of opening a Session, loading all manifests that match specific conditions into the cache. This means when it is time to read a chunk, the manifest will already be in the cache and the chunk can be found without having to first load the manifest from storage.
+
+### Configuring preloading
+
+To configure manifest preloading, you can use the `ManifestPreloadConfig` class to specify the `ic.RepositoryConfig.manifest.preload` field.
+
+```python exec="on" session="perf" source="material-block"
+preload_config = ic.ManifestPreloadConfig(
+    preload_if=ic.ManifestPreloadCondition.name_matches("x"),  # preload all manifests that match the regex "x"
```

Comment:
I think name_matches takes a regex, so this may match much more than they want? I don't remember if we do full match or we match anywhere in the name. So, this may need something like `("^x$")`. But I'd have to check how we match it, maybe you already did. It would be nice to add it to this docs.

Reply:
You are totally right, I'll refine this

Reply:
done